File: app/src/main/java/com/yousef/ta3leem/MainActivity.java
package com.yousef.ta3leem;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.yousef.ta3leem.Data.FireBase.CallBacks.allStudentsFireBaseCallBack;
import com.yousef.ta3leem.Data.FireBase.CallBacks.allTeachersFirebaseCallBack;
import com.yousef.ta3leem.Data.FireBase.FireBaseHelper.Student;
import com.yousef.ta3leem.Data.FireBase.FireBaseHelper.Teacher;
import com.yousef.ta3leem.Repository.Repo;

import java.util.List;

public class MainActivity extends AppCompatActivity {
    Repo repo = new Repo();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_app);
        getAllStudents();
        getAllTeachers();
    }

    /**
     * Cache the firebase on the client app for faster data upload
     */
    public void getAllStudents(){
        repo.getStudentsOneTime(new allStudentsFireBaseCallBack() {
            @Override
            public void onCallBack(List<Student> allStudents) {

            }
        });
    }

    public void getAllTeachers(){
        repo.getTeachersOneTime(new allTeachersFirebaseCallBack() {
            @Override
            public void allTeachersOnCallBack(List<Teacher> allTeachers) {

            }
        });
    }
}

File: app/src/main/java/com/yousef/ta3leem/Kotlin/ui/MainActivity.kt
package com.yousef.ta3leem.Kotlin.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.NavController
import androidx.navigation.findNavController
import com.yousef.ta3leem.Kotlin.Model.ChatUser
import com.yousef.ta3leem.Kotlin.ui.Login.LoginFragmentDirections
import com.yousef.ta3leem.R
import io.getstream.chat.android.client.ChatClient

class MainActivity : AppCompatActivity() {

    private lateinit var navController: NavController
    private val client = ChatClient.instance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_app)

        navController = findNavController(R.id.navHostFragment)

        if (navController.currentDestination?.label.toString().contains("login")) {
            val currentUser = client.getCurrentUser()
            if (currentUser != null) {
                val user = ChatUser(currentUser.name, currentUser.id)
                val action = LoginFragmentDirections.actionLoginFragmentToChannelFragment(user)
                navController.navigate(action)
            }
        }

    }
}
